File: 20220913/ompp-0.8.5/opari/tool/Test/test4.c
#include <stdio.h>

int main() {
  int i;

  #pragma omp parallel
  {
    printf("parallel\n");
    #pragma omp for nowait
    for (i=0; i<4; ++i) printf("do ; ; ; nowait %d\n", i);

    #pragma omp barrier

    #pragma omp \
	   /* unnütz */ \
	   /* mehr unnütz */ \
	   for /* KEIN nowait */
    for (i=0; i<4; ++i) {
      printf("do nowait? %d\n", i);
    }

    #pragma omp for
    for (i=0; i<4; ++i) {
      printf("do %d\n", i);
    }
  }
}

File: 20220913/ompp-0.8.5/opari/tool/Test/test4.f90
       program test4
       integer i

       !$omp parallel
       write(*,*) "parallel"
         !$omp do
         do i=1,4
           write(*,*) "do nowait",i
         enddo
         !$omp enddo nowait

         !$omp barrier

         !$omp do
         do i=1,4
           write(*,*) "do nowait?",i
         enddo
         !$omp   end   &  ! continue
         !$omp &       &  ! still continue
         !$omp & do       ! kein nowait

         !$omp do
         do i=1,4
           write(*,*) "do",i
         enddo
         !$omp end do
       !$omp end parallel

       end
